.github/workflows/wheels-test.yml

```diff
@@ -152,11 +168,48 @@ jobs:
         date: ${{ inputs.date }}
         sha: ${{ inputs.sha }}
 
-    - name: Set CTK-related vars from input CTK versions
-      uses: rapidsai/shared-action-workflows/wheel-ctk-name-gen@branch-23.06
-      with:
-        ctk: ${{ matrix.ctk }}
-        package-name: ${{ inputs.package-name }}
-
     - name: Run citestwheel
-      run: /citestwheel.sh
+      run: |
+        set -eoxu pipefail
+
+        ctk_tag="${{ matrix.ctk }}"
+        ctk_major=$(echo $ctk_tag | cut -d'.' -f1)
+        cuda_suffix="cu${ctk_major}"
+        wheelname="${{ inputs.package-name }}_cu${ctk_major}"
+
+        export RAPIDS_PY_WHEEL_CUDA_SUFFIX=${cuda_suffix}
+        export RAPIDS_PY_WHEEL_NAME=${wheelname}
+
+        export RAPIDS_BEFORE_TEST_COMMANDS_AMD64="${{ inputs.test-before-amd64 }}"
+        export RAPIDS_BEFORE_TEST_COMMANDS_ARM64="${{ inputs.test-before-arm64 }}"
+
+        mkdir -p ./dist
+
+        arch=$(uname -m)
+
+        # need this to init pyenv first
+        eval "$(pyenv init -)"
+
+        # use pyenv to set appropriate python as default before citestwheel
+        pyenv global "${RAPIDS_PY_VERSION}" && python --version
+
+        rapids-download-wheels-from-s3 ./dist
+
+        if [ "${arch}" == "x86_64" ]; then
+                sh -c "${RAPIDS_BEFORE_TEST_COMMANDS_AMD64}"
+        elif [ "${arch}" == "aarch64" ]; then
+                sh -c "${RAPIDS_BEFORE_TEST_COMMANDS_ARM64}"
+        fi
+
+        # see: https://cibuildwheel.readthedocs.io/en/stable/options/#test-extras
+        extra_requires_suffix=''
+        if [ "${CIBW_TEST_EXTRAS}" != "" ]; then
+                extra_requires_suffix="[${CIBW_TEST_EXTRAS}]"
+        fi
```

Comment:
Also perhaps out of scope for this PR, but I'd like to for RAPIDS packages to expose the `[test]` extra as a matter of policy so that this just works without inputs.